### PR TITLE
Enable the master replacement link feature in python

### PR DESCRIPTION
**Pull request by sima-zhu**
This is a follow up step of master link replacement work.

Tested in template: 
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=530466&view=logs&j=3c922a7d-1305-5297-ba3f-884e0ad84366&t=3c922a7d-1305-5297-ba3f-884e0ad84366

Retrieved release tag successfully:
image

### CODE REVIEW

File: eng/pipelines/templates/steps/build-artifacts.yml [hunk 1]
```diff
@@ -18,13 +18,6 @@ steps:
   - script: |
       pip install -r eng/ci_tools.txt
     displayName: 'Prep Py2 Environment'
-    
-  - template: /eng/common/pipelines/templates/steps/replace-relative-links.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/build-artifacts.yml:22]: Should wait until we fix relative links in the repo.

File: eng/pipelines/templates/stages/archetype-python-release.yml [hunk 2]
```diff
@@ -134,6 +134,7 @@ stages:
                           BlobSASKey: '$(azure-sdk-docs-prod-sas)'
                           BlobName: '$(azure-sdk-docs-prod-blob-name)'
                           TargetLanguage: 'python'
+                          ArtifactLocation: '$(Pipeline.Workspace)/${{parameters.ArtifactName}}'
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-python-release.yml:137]: Shouldn't this be `$(Pipeline.Workspace)/${{artifact.safeName}}`?

Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-python-release.yml:137]: We put Doc files in `$(Pipeline.Workspace)/${{artifact.safeName}}`, but the source code which we parse the artifact metadata from is inside of `packages` folder, (aka  `ArtifactName`).

**Review by weshaggard — APPROVED** [review]
Please wait for releases to finish.

**Pull request merged**